1.8/administration/logging/elk.md

```diff
@@ -1,147 +1,200 @@
-                -u dcos-spartan.service                  \
-                -u dcos-link-env.service                 \
-                -u dcos-vol-discovery-priv-agent.service \
-                -u dcos-logrotate-agent.service          \
-                | ./logstash-forwarder -config logstash.conf
-
-3.  Run `logstash.sh` as a service.
-
-    **Important:** If any of these DC/OS services are restarted, you must restart this script.
+1. Create a script that will parse the output of the DC/OS Agent `journalctl` logs and will funnel them all to `/var/log/dcos/dcos/dcos.log`.
+
+```bash
+sudo tee /etc/systemd/system/dcos-journalctl-filebeat.service<<-EOF 
+[Unit]
+Description=DCOS journalctl parser to filebeat
+Wants=filebeat.service
+After=filebeat.service
+
+[Service]
+Restart=always
+RestartSec=5
+ExecStart=/bin/sh -c '/usr/bin/journalctl --no-tail -f      \
+  -u dcos-3dt.service                      \
+  -u dcos-logrotate-agent.timer            \
+  -u dcos-3dt.socket                       \
+  -u dcos-mesos-slave.service              \
+  -u dcos-adminrouter-agent.service        \
+  -u dcos-minuteman.service                \
+  -u dcos-adminrouter-reload.service       \
+  -u dcos-navstar.service                  \
+  -u dcos-adminrouter-reload.timer         \
+  -u dcos-rexray.service                   \
+  -u dcos-cfn-signal.service               \
+  -u dcos-setup.service                    \
+  -u dcos-download.service                 \
+  -u dcos-signal.timer                     \
+  -u dcos-epmd.service                     \
+  -u dcos-spartan-watchdog.service         \
+  -u dcos-gen-resolvconf.service           \
+  -u dcos-spartan-watchdog.timer           \
+  -u dcos-gen-resolvconf.timer             \
+  -u dcos-spartan.service                  \
+  -u dcos-link-env.service                 \
+  -u dcos-vol-discovery-priv-agent.service \
+  -u dcos-logrotate-agent.service          \
+  > /var/log/dcos/dcos.log 2>&1'
+ExecStartPre=/usr/bin/journalctl --vacuum-size=10M
+
+[Install]
+WantedBy=multi-user.target
+EOF
+```
+
+# <a name="all-3"></a>Step 3: All Nodes
+
+1. In all nodes, start and enable the filebeat log parsing services created above:
+
+```bash
+sudo chmod 0755 /etc/systemd/system/dcos-journalctl-filebeat.service
+sudo systemctl daemon-reload
+sudo systemctl start dcos-journalctl-filebeat.service
+sudo chkconfig dcos-journalctl-filebeat.service on
+sudo systemctl start filebeat
+sudo chkconfig filebeat on
+```
+
+# <a name="all"></a>Step 3: ELK Node Notes
+
+The ELK stack will receive, store, search and display information about the logs parsed by the Filebeat instances configured above for all nodes in the cluster. In this example, we are streaming directly from Filebeat into ElasticSearch, so Logstash is not used in this architecture. If you're interested in filtering, parsing and grok'ing the logs with an intermediate Logstash stage, please check the Logstash [documentation][8].
```

Comment:
Suggest: `In this document, we are streaming directly from Filebeat into ElasticSearch, so Logstash is not used in this architecture. `